### PR TITLE
Issue 6599: Fix flaky operationlog metric tests

### LINKED ISSUE #6599: Fix flaky SS OperationLogMetrics test
**Describe the bug**
The operationLog metrics test seems to be flaky. The r0.11 branch is where it seems to have highest failing occurrences .This flakiness of the test needs to be fixed.
test:-  [io.pravega.segmentstore.server.SegmentStoreMetricsTests.testOperationProcessorMetrics]

**To Reproduce**
Run Unit tests and we should see the above test failing intermittently.

**Screenshots**
N/A

**Additional information**
N/A

### PR DESCRIPTION
**Pull request by abhinb**
**Change log description**  
This PR fixes the flakiness in the OperationLogMetric tests, specifically the failure due to operationlog size gauge metric in the "io.pravega.segmentstore.server.SegmentStoreMetricsTests.testOperationProcessorMetrics".  

**Purpose of the change**  
Fixes #6599 

**What the code does**  
The code as part of the fix makes sure that the gauge metric created is removed from the metric cache when a container is closed. This removal of metric from the internal cache makes sure that effects from other tests do not affect the current run by not retrieving stale values. 
When we create/report a gauge metric using DynamicLoggerImpl.java, it records the metric in an internal cache that it maintains. When we close a container we need to make sure that we freeze the gauge value( that is no more updates to this metric since container is shutting down). Freezing the gauge results in removal of this gauge metric from the internal cache.
Failure to do so can result in stale values being retrieved during the test runs causing flakiness in the SegmentStoreMetricsTests.testOperationProcessorMetrics.

**How to verify it**  
All tests should pass.

### CODE REVIEW

File: segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentStoreMetrics.java [hunk 1]
```diff
@@ -280,6 +281,15 @@ public void operationsFailed(Collection<CompletableOperation> operations) {
                 GLOBAL_OPERATION_LATENCY.reportFailValue(millis);
             }
         }
+
```
Comment (RaulGracia) [inline segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentStoreMetrics.java:284]: Is this change necessary?

File: segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DurableLog.java [hunk 2]
```diff
@@ -221,7 +220,7 @@ private boolean performRecovery() {
             RecoveryProcessor p = new RecoveryProcessor(this.metadata, this.durableDataLog, this.memoryStateUpdater);
             int recoveredItemCount = p.performRecovery();
             this.operationProcessor.getMetrics().operationsCompleted(recoveredItemCount, timer.getElapsed());
-            SegmentStoreMetrics.reportOperationLogSize(recoveredItemCount, this.getId());
+            this.operationProcessor.getMetrics().reportOperationLogSize(recoveredItemCount, this.getId());
```
Comment (RaulGracia) [inline segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DurableLog.java:223]: If this is returning a `SegmentStoreMetrics.OperationProcessor` object, why `reportOperationLogSize` is defined as `static`?

**Review by RaulGracia — CHANGES_REQUESTED** [review]

File: segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DurableLog.java (hunk 2 above)
Comment (abhinb) [reply segmentstore/server/src/main/java/io/pravega/segmentstore/server/logs/DurableLog.java:223]: will correct this(had just moved the method)

**Comment by abhinb** [comment]
tpc/4380 has passed

**Comment by RaulGracia** [comment]
@abhinb have you checked that this metrics is being correctly reported in Grafana?

**Comment by abhinb** [comment]
@RaulGracia  I checked the metric with the changes in this PR. The metric is reported correctly in Grafana.

File: segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentStoreMetrics.java (hunk 1 above)
Comment (RaulGracia) [reply segmentstore/server/src/main/java/io/pravega/segmentstore/server/SegmentStoreMetrics.java:284]: Well, what I see in the diff is that you are moving `reportOperationLogSize()` from line 519 to line 284. My question is whether moving this method is necessary or not.

**Review by RaulGracia — APPROVED** [review]

**Pull request merged**